### PR TITLE
05.01 - 05.05

### PR DESCRIPTION
**Pull request by dijeth**

### CODE REVIEW

**Comment by jsru** [comment]
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

**Comment by jsru** [comment]
Добавляю преподавателя (@ShGKme) для код-ревью.

File: 05-vue-router/05-AuthGuard/router/index.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  await router.push({ name: 'login', query: { ...to.query, from: to.path } });
+};
+
+const guest: TMiddleware = async ({ router }) => {
+  if (!isAuthenticated()) {
+    return;
+  }
+
+  await router.replace({ name: 'index' });
+};
```
Comment (ShGKme) [inline 05-vue-router/05-AuthGuard/router/index.ts:31]: In my opinion, it's better to keep middlewares implementing the default interface of router hooks - `to, from` as params and returned value that can be directly used as value for VueRouter hook. 

Then we can use middleware as a general hook outside of our custom middleware and keep the interface is known to any Vue developer.

I know, that it is from the Markus article. Though, in new VueRouter version with async hooks it is easy to implement.

**Review by ShGKme — APPROVED** [review]
All tasks approved 👍

**Pull request merged**